components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/FileBasedApplicationDAO.java

```diff
@@ -57,6 +58,20 @@ public ServiceProvider getApplication(String applicationName, String tenantDomai
         return ApplicationManagementServiceComponent.getFileBasedSPs().get(applicationName);
     }
 
+    @Override
+    public LocalAndOutboundAuthenticationConfig getApplicationLocalAndOutboundAuthenticationConfig
+            (String applicationName, String tenantDomain) throws IdentityApplicationManagementException {
+
+        throw new IdentityApplicationManagementException("Not supported in file based dao.");
```

Comment:
can you confirm that there is no impact in not implementing this?

Reply:
I think we don't need to provide support for file-based SPs, cause in the console we don't have support for filebased SPs and this implementation is developed from the method `getApplicationExcludingFileBasedSPs`. By not implementing this I don't see any problems. Are we going to support file-based SPs in the future?

But current methods exposed via service is

- getApplicationClaimConfig
- getApplicationLocalAndOutboundAuthenticationConfig

by considering the name one could deduct that this support filebasedSPs, hence I will change the method name to

- getApplicationClaimConfigExcludingFileBasedSPs
- getApplicationLocalAndOutboundAuthenticationConfigExcludingFileBasedSPs

WDYT?

Reply:
+1